engine/access/rpc/backend/backend_transactions.go

```diff
@@ -245,15 +245,34 @@ func (b *backendTransactions) DeriveTransactionStatus(
 		if err != nil {
 			return flow.TransactionStatusUnknown, err
 		}
+		refHeight := referenceBlock.Height
 		// get the latest finalized block from the state
 		finalized, err := b.state.Final().Head()
 		if err != nil {
 			return flow.TransactionStatusUnknown, err
 		}
+		finalizedHeight := finalized.Height
 
-		// Have to check if finalized height is greater than reference block height rather than rely on the subtraction, since
-		// heights are unsigned ints
-		if finalized.Height > referenceBlock.Height && finalized.Height-referenceBlock.Height > flow.DefaultTransactionExpiry {
+		// the last full height is the height where we have recieved all
```

Comment:
we can add a unit testcase for this - I can do it as a separate issue if you would like

Reply:
done in https://github.com/onflow/flow-go/pull/390/commits/dc40f7e8af314577323d5d32607807937384a5d5